### PR TITLE
Contributions drawer in dashboard

### PR DESCRIPTION
**Pull request by hdiniz**
:warning: Requires https://github.com/opencollective/opencollective-api/pull/10048
Related https://github.com/opencollective/opencollective/issues/7345

# Description

Adds a preview feature of the new contribution management in dashboard. A rough draft of the contribution drawer details, previous charges and timeline.

# Screenshots

image
image
image

### CODE REVIEW

File: components/dashboard/sections/contributions/Contributions.tsx [hunk 1]
```diff
@@ -444,72 +634,188 @@ const Contributions = ({ accountSlug, direction }: ContributionsProps) => {
...
+          <EditOrderModal
+            accountSlug={accountSlug}
+            order={editOrder.order}
+            action={editOrder.action}
+            onClose={() => setEditOrder({ order: null, action: null })}
           />
-        </div>
+        )}
+      </div>
+      {LoggedInUser?.hasPreviewFeatureEnabled(PREVIEW_FEATURE_KEYS.DASHBOARD_CONTRIBUTION_DETAILS) && (
```
Comment (kewitz) [inline components/dashboard/sections/contributions/Contributions.tsx:802]: I don't get this check. Will the drawer just fail to open if the preview feature is off?

File: components/Link.js [hunk 2]
```diff
@@ -72,4 +75,8 @@ class Link extends React.Component {
   }
 }
 
-export default Link;
+/**
+ * @typedef {{ target?: string, href?: string | Record<string, any>, animate?: any, className?: string, title?: string, onClick?: () => void, openInNewTab?: boolean, className?: string, children?: any, 'data-cy'?: string  } & React.HTMLAnchorElement} LinkProps
+ * @type React.ForwardRefRenderFunction<HTMLAnchorElement, LinkProps>
+ */
+export default React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);
```
[the remark this answers is not in the document]
Comment (hdiniz) [reply components/Link.js:82]: Can you give me an example of what you mean? If you are refering to the name of the variable, the linter complains if I call it `ref`

[the remark this answers is not in the document]
Comment (kewitz) [reply components/Link.js:82]: I meant implementing forwardRef in the component itself and not like a wrapper.

File: components/dashboard/sections/contributions/Contributions.tsx [hunk 3]
```diff
@@ -130,21 +227,70 @@ const dashboardContributionsQuery = gql`
...
+              />
+            </div>
+          </div>
+          <div className="overflow-hidden">
+            <div className="overflow-hidden text-ellipsis whitespace-nowrap text-sm font-medium leading-5">
+              {toAccount.slug || toAccount.name}
+            </div>
+            <div className="overflow-hidden text-ellipsis whitespace-nowrap text-xs leading-4">
+              {fromAccount.slug || fromAccount.name}
+            </div>
```
Comment (kewitz) [inline components/dashboard/sections/contributions/Contributions.tsx:268]: I think we should prioritize name here, it looks especially weird with vendors.
```suggestion
            <div className="overflow-hidden text-ellipsis whitespace-nowrap text-sm font-medium leading-5">
              {toAccount.name || toAccount.slug}
            </div>
            <div className="overflow-hidden text-ellipsis whitespace-nowrap text-xs leading-4">
              {fromAccount.name || fromAccount.slug}
            </div>
```

File: components/dashboard/sections/contributions/Contributions.tsx [hunk 4]
```diff
@@ -130,21 +227,70 @@ const dashboardContributionsQuery = gql`
...
-        </LinkCollective>
+        <AccountHoverCard
+          account={toAccount}
+          trigger={
+            <div className="flex items-center gap-2">
+              <div>
+                <Avatar size={32} collective={toAccount} displayTitle={false} />
+              </div>
+              <div className="overflow-hidden text-ellipsis whitespace-nowrap text-sm leading-5">
+                {toAccount.slug || toAccount.name}
```
Comment (kewitz) [inline components/dashboard/sections/contributions/Contributions.tsx:289]: Same here.
```suggestion
                {toAccount.name || toAccount.slug}
```

File: components/dashboard/sections/contributions/Contributions.tsx [hunk 5]
```diff
@@ -154,32 +300,35 @@ const getColumns = ({ tab, setEditOrder, intl, isIncoming }) => {
...
-        </LinkCollective>
+        <AccountHoverCard
+          account={fromAccount}
+          trigger={
+            <div className="flex items-center gap-2">
+              <div>
+                <Avatar size={32} collective={fromAccount} displayTitle={false} />
+              </div>
+              <div className="overflow-hidden text-ellipsis whitespace-nowrap text-sm leading-5">
+                {fromAccount.slug || fromAccount.name}
```
Comment (kewitz) [inline components/dashboard/sections/contributions/Contributions.tsx:311]: Same here.
```suggestion
                {fromAccount.name || fromAccount.slug}
```

File: components/dashboard/sections/contributions/Contributions.tsx (hunk 3 above)
Comment (kewitz) [reply components/dashboard/sections/contributions/Contributions.tsx:268]: image

**Review by kewitz — APPROVED** [review]
This is awesome! Great work! Feel free to ship this.

**Pull request merged**